numpy/distutils/armccompiler.py

```diff
@@ -0,0 +1,28 @@
+from __future__ import division, absolute_import, print_function               
+
+from distutils.unixccompiler import UnixCCompiler                              
+
+class ArmCCompiler(UnixCCompiler):
+
+    """
+    Arm compiler.
+    """
+
+    compiler_type = 'arm'
+    cc_exe = 'armclang'
+    cxx_exe = 'armclang++'
+
+    def __init__(self, verbose=0, dry_run=0, force=0):
+        UnixCCompiler.__init__(self, verbose, dry_run, force)
+        cc_compiler = self.cc_exe
+        cxx_compiler = self.cxx_exe
+        self.set_executables(compiler=cc_compiler +
+                                      ' -mcpu=native -O3 -fPIC',
```

Comment:
```suggestion
                                      ' -O3 -fPIC',
```
default settings should target generic builds, isn't it?

flag `-fPIC` is supposed to be defined by `python/sysconfig`.
flag `-O3` shouldn't be the default, and also optimization level should be left to be configured by `python/sysconfig`.

Reply:
Ah okay - makes sense but just checked against another compiler setup such as [intelccompiler.py](https://github.com/ARCCA/numpy/blob/1cfdac82ac793061d8ca4b07c046fc6b21ee7e54/numpy/distutils/intelccompiler.py) where `-fPIC and -O3 is defined in the compiler definition:

https://github.com/ARCCA/numpy/blob/1cfdac82ac793061d8ca4b07c046fc6b21ee7e54/numpy/distutils/intelccompiler.py#L21-L23

Happy to remove it if there is a better way of specifying compiler options.

Reply:
`-O2` or `-O3` is common to include, as is `-fPIC`. It has always been added in `numpy.distutils`, not sure if now `sysconfig` does that and we get duplicates. Good build systems have a CLI/API to propagate optimization settings across builds, but unfortunately `distutils` is not a good build system, so we tended to hardcode it. Should be okay to be pragmatic here.

Agree with removing `-mcpu=native` though.

Reply:
according to the provided build log in https://github.com/numpy/numpy/pull/20416#issuecomment-975951194, `-fPIC` is getting duplicated.
```Bash
C compiler: /snx11273/home/username/hackathon/spack/lib/spack/env/arm/armclang -Wno-unused-result -Wsign-compare -Wunreachable-code -DNDEBUG -g -fwrapv -O3 -Wall -fPIC -fPIC
```
but it's kinda weird for not seeing other optimization flag or even flag `-mcpu=native`, it seems this build was using `unixcompiler` and the arm compiler was specified via env variable `CC`. @tomgreen66, Have you used build option  `--compiler=arm` in that build?

Reply:
Good spot - I used Spack to build the example and it only uses fcompiler and as you point out it seems to not specify the compiler so must use the default UnixCompiler (Spack wraps all compilers/linkers so it can add its own flags). I had tested without Spack but have now attached non-Spack build. Should Spack use the numpy compiler configs or just override the UnixCompiler?
[compile.log](https://github.com/numpy/numpy/files/7663991/compile.log)

Reply:
> Should Spack use the numpy compiler configs or just override the UnixCompiler?

it wouldn't make difference as long as the same compiler executables, and flags are used.

I think we're good here, just need to remove the flag `-mcpu=native`.

Reply:
Hi - just updated brand to remove `-mcpu=native` - thanks for all the advice.

Reply:
From the log in the comment above, I now see
```
C compiler: /snx11273/<path>/bin/armclang \
    -pthread -Wno-unused-result -Wsign-compare -Wunreachable-code -DNDEBUG \
    -g -fwrapv -O3 -Wall -fPIC -fPIC -ftrapping-math
```

where <path> ends in `arm-linux-compiler-21.0_Generic-AArch64_RHEL-7_aarch64-linux`

The duplication of `-fPIC` is harmless.